cmd/talosctl/cmd/talos/reboot.go

```diff
@@ -21,8 +21,20 @@ var rebootCmd = &cobra.Command{
 	Args:  cobra.NoArgs,
 	RunE: func(cmd *cobra.Command, args []string) error {
 		return WithClient(func(ctx context.Context, c *client.Client) error {
-			if err := c.Reboot(ctx); err != nil {
-				return fmt.Errorf("error executing reboot: %s", err)
+			mode, err := cmd.Flags().GetInt("mode")
+			if err != nil {
+				return fmt.Errorf("error getting input value for --mode flag: %s", err)
+			}
+
+			switch mode {
```

Comment:
this can rewritten to avoid duplication:

```go
opts := []client.RebootOption{}

if mode == "powercycle" {
   opts = append(opts, client.WithPowerCycle
}

if err := c.Reboot(ctx, opts...); err != nil {
    return fmt.Errorf(....)
}
```
```